kernel/nano_core/src/lib.rs

```diff
@@ -16,125 +16,63 @@
 //! 
 
 #![no_std]
+#![no_main]
+#![feature(naked_functions)]
 
-#[macro_use] extern crate log;
-extern crate alloc;
-extern crate spin;
-extern crate multiboot2;
-extern crate kernel_config; // our configuration options, just a set of const definitions.
-extern crate irq_safety; // for irq-safe locking and interrupt utilities
-extern crate logger;
-extern crate state_store;
-extern crate memory; // the virtual memory subsystem
-extern crate no_drop;
-extern crate stack;
-extern crate serial_port_basic;
-extern crate mod_mgmt;
-extern crate exceptions_early;
-#[macro_use] extern crate vga_buffer;
-extern crate panic_entry; // contains required panic-related lang items
-#[cfg(not(loadable))] extern crate captain;
-extern crate memory_initialization;
-extern crate boot_info;
-
+extern crate panic_entry;
 
 use core::ops::DerefMut;
 use memory::VirtualAddress;
 use kernel_config::memory::KERNEL_OFFSET;
-use serial_port_basic::{take_serial_port, SerialPortAddress};
-use boot_info::BootInformation;
+
+mod util;
+cfg_if::cfg_if! {
+    if #[cfg(feature = "bios")] {
+        mod bios;
+    } else if #[cfg(feature = "uefi")] {
+        mod uefi;
+    }
+    // If neither is enabled, and build script wolud have already emitted an
```

Comment:
might as well throw a `compile_error!()` here anyway just in case some noob modifies the build script accidentally